core-java-modules/core-java-regex/src/test/java/com/baeldung/regex/noncapturinggroups/NonCapturingGroupUnitTest.java

```diff
@@ -0,0 +1,135 @@
+package com.baeldung.regex.noncapturinggroups;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NonCapturingGroupUnitTest {
+
+    private static final Pattern SIMPLE_URL_PATTERN = Pattern.compile("[^:]+://(?:[.a-z]+/?)+");
+    private static final Pattern SCOPED_CASE_INSENSITIVE_URL_PATTERN = Pattern.compile("[^:]+://(?i:[.a-z]+/?)+");
+    private static final Pattern SIMPLE_URL_PATTERN_WITH_SPECIFIC_ENDING_PATH = Pattern.compile("[^:]+://(?:[.a-z]+/?)+?/ending-path");
+    private static final Pattern SCOPED_CASE_INSENSTIIVE_URL_PATTERN_WITH_ENDING_PATH = Pattern.compile("[^:]://(?i:[.a-z]+/?)+?/ending-path");
+    private static final Pattern CASE_INSENSITIVE_URL_PATTERN = Pattern.compile("[^:]+://(?:[.a-z]+/?)+?(/ending-path)", Pattern.CASE_INSENSITIVE);
+    private static final Pattern SCOPED_CASE_SENSITIVE_URL_PATTERN = Pattern.compile("[^:]+://(?-i:[.a-z]+/?)+?(/ending-path)", Pattern.CASE_INSENSITIVE);
+    private static final Pattern INDEPENDENT_URL_PATTERN_WITH_ENDING_PATH = Pattern.compile("[^:]+://(?>[.a-z]+/?)+/ending-path");
+    private static final Pattern INDEPENDENT_URL_PATTERN_WITH_ENDING_PATH_AND_BACKTRACKING = Pattern.compile("[^:]+://(?>(?:[.a-z]+/?)+/)ending-path");
+
+    @Test
+    void givenSimpleUrlPattern_whenValidUrlProvided_thenMatches() {
+        Matcher urlMatcher = SIMPLE_URL_PATTERN.matcher("http://www.microsoft.com/");
+        
+        Assertions.assertThat(urlMatcher.matches())
+            .isTrue();
```

Comment:
two space indents for line continuations. For the isTrue/False, I would just put it on the same line

Reply:
Ok. I can get this updated tonight.

Reply:
I've updated the PR with the requested changes.